test/go-tests/test_proxyauth.go

```diff
@@ -0,0 +1,125 @@
+          tasks:
+            - name: "deployment"
+              properties:
+                deploymentstrategy: "blue_green_service"
+            - name: "test"
+              properties:
+                teststrategy: "performance"
+            - name: "evaluation"
+            - name: "release"
+        - name: "rollback"
+          triggeredOn:
+            - event: "prod.delivery.finished"
+              selector:
+                match:
+                  result: "fail"
+          tasks:
+            - name: "rollback"
+
+        - name: "delivery-direct"
+          triggeredOn:
+            - event: "dev.delivery-direct.finished"
+          tasks:
+            - name: "deployment"
+              properties:
+                deploymentstrategy: "direct"
+            - name: "release"
+`
+
+func Test_ProxyAuth(t *testing.T) {
+	repoLocalDir := "../assets/podtato-head"
+	projectName := "proxy-auth"
+	serviceName := "helloservice"
+	secondServiceName := "helloservice2"
+	serviceChartLocalDir := path.Join(repoLocalDir, "helm-charts", "helloservice.tgz")
+	serviceJmeterDir := path.Join(repoLocalDir, "jmeter")
+
+	t.Logf("Creating a new project %s with Gitea Upstream", projectName)
+	shipyardFilePath, err := CreateTmpShipyardFile(testingProxyShipyard)
+	require.Nil(t, err)
+	projectName, err = CreateProjectWithProxy(projectName, shipyardFilePath)
+	require.Nil(t, err)
+
+	t.Logf("Creating service %s in project %s", serviceName, projectName)
+	_, err = ExecuteCommandf("keptn create service %s --project %s", serviceName, projectName)
+	require.Nil(t, err)
+
+	t.Logf("Adding resource for service %s in project %s", serviceName, projectName)
+	_, err = ExecuteCommandf("keptn add-resource --project %s --service=%s --all-stages --resource=%s --resourceUri=%s", projectName, serviceName, serviceChartLocalDir, "helm/helloservice.tgz")
+	require.Nil(t, err)
+
+	t.Log("Adding jmeter config in prod")
+	_, err = ExecuteCommandf("keptn add-resource --project=%s --service=%s --stage=%s --resource=%s --resourceUri=%s", projectName, serviceName, "prod", serviceJmeterDir+"/jmeter.conf.yaml", "jmeter/jmeter.conf.yaml")
+	require.Nil(t, err)
+
+	t.Log("Adding load test resources for jmeter in prod")
+	_, err = ExecuteCommandf("keptn add-resource --project=%s --service=%s --stage=%s --resource=%s --resourceUri=%s", projectName, serviceName, "prod", serviceJmeterDir+"/load.jmx", "jmeter/load.jmx")
+	require.Nil(t, err)
+
+	t.Logf("Trigger delivery of helloservice:v0.1.0")
+	_, err = ExecuteCommandf("keptn trigger delivery --project=%s --service=%s --image=%s --tag=%s --sequence=%s", projectName, serviceName, "ghcr.io/podtato-head/podtatoserver", "v0.1.0", "delivery")
+	require.Nil(t, err)
+
+	t.Logf("Updating project credentials")
+	user := GetGiteaUser()
+	token, err := GetGiteaToken()
+	require.Nil(t, err)
+
+	namespace := osutils.GetOSEnvOrDefault(KeptnNamespaceEnvVar, DefaultKeptnNamespace)
+
+	squidIP, err := GetSquidExternalIP(namespace)
```

Comment:
is it somehow possible (e.g. by inspecting the logs of the proxy) to verify that:
1. The resource service is indeed going through the proxy for requests regarding this project
2. The resource service does not go through the proxy for other projects where no proxy is configured

Reply:
will check

Reply:
Actually we have no logs from proxy, so there is no way to test if the routing goes through it